library/Opus/Search/Query.php

```diff
@@ -0,0 +1,479 @@
+	 * @param Opus_Search_Filter_Base $subFilter filter to be satisfied by all matching documents in addition
+	 * @return $this fluent interface
+	 */
+	public function setSubFilter( $name, Opus_Search_Filter_Base $subFilter ) {
+		if ( !is_string( $name ) || !$name ) {
+			throw new InvalidArgumentException( 'invalid sub filter name' );
+		}
+
+		if ( !is_array( $this->_data['subfilters'] ) ) {
+			$this->_data['subfilters'] = array( $name => $subFilter );
+		} else {
+			$this->_data['subfilters'][$name] = $subFilter;
+		}
+
+		return $this;
+	}
+
+	/**
+	 * Removes some previously defined subfilter from current query again.
+	 *
+	 * @note This isn't affecting server-side caching of selected filter but
+	 *       reverting some parts of query compiled on client-side.
+	 *
+	 * @see Opus_Search_Query::setSubFilter()
+	 *
+	 * @param string $name name of filter to remove from query again
+	 * @return $this fluent interface
+	 */
+	public function removeSubFilter( $name ) {
+		if ( !is_string( $name ) || !$name ) {
+			throw new InvalidArgumentException( 'invalid sub filter name' );
+		}
+
+		if ( is_array( $this->_data['subfilters'] ) ) {
+			if ( array_key_exists( $name, $this->_data['subfilters'] ) ) {
+				unset( $this->_data['subfilters'][$name] );
+			}
+
+			if ( !count( $this->_data['subfilters'] ) ) {
+				$this->_data['subfilters'] = null;
+			}
+		}
+
+		return $this;
+	}
+
+	/**
+	 * Retrieves named map of subfilters to include on querying search engine.
+	 *
+	 * @return Opus_Search_Filter_Base[]
+	 */
+	public function getSubFilters() {
+		return $this->_data['subfilters'];
+	}
+
+	public static function getParameterDefault( $name, $fallbackIfMissing, $oldName = null ) {
+		$config   = Opus_Search_Config::getDomainConfiguration();
+		$defaults = $config->parameterDefaults;
+
+		if ( $defaults instanceof Zend_Config ) {
+			return $defaults->get( $name, $fallbackIfMissing );
+		}
+
+		if ( $oldName ) {
+			return $config->get( $oldName, $fallbackIfMissing );
+		}
+
+		return $fallbackIfMissing;
+	}
+
+	/**
+	 * Retrieves configured default offset for paging results.
+	 *
+	 * @return int
+	 */
+	public static function getDefaultStart() {
+		return static::getParameterDefault( 'start', 0 );
+	}
+
+	/**
+	 * Retrieves configured default number of rows to show (per page).
+	 *
+	 * @return int
+	 */
+	public static function getDefaultRows() {
+		return static::getParameterDefault( 'rows', 10, 'numberOfDefaultSearchResults' );
+	}
+
+	/**
+	 * Retrieves configured default sorting.
+	 *
+	 * @return string[]
+	 */
+	public static function getDefaultSorting() {
+		$sorting = static::getParameterDefault( 'sortField', 'score desc' );
+
+		$parts = preg_split( '/[\s,]+/', trim( $sorting ), null, PREG_SPLIT_NO_EMPTY );
+
+		$sorting = array( array_shift( $parts ) );
+
+		if ( !count( $parts ) ) {
+			$sorting[] = 'asc';
+		} else {
+			$dir = array_shift( $parts );
+			if ( strcasecmp( $dir, 'asc' ) || strcasecmp( $dir, 'desc' ) ) {
```

Comment:
You are right. This code is considering ASC to be default sorting order 
if there is no explicit definition. I considered ASC to be a somewhat 
usual default e.g. for sorting by title or name of author or year of 
publication (but according to your post it isn't that usual in Opus).

The method actually reads an optionally given parameter named 
"sortField". If it is missing, it processes the default "score desc". 
Processing includes splitting string into chunks of whitespace-separated 
parts taking first part to be the name of property to be sorted. Second 
part is considered optional and in processing string it is defaulting to 
"ASC" ... which isn't the case if reading parameter has failed before 
using "score desc". If second part is given it is expected to be either 
desc or asc.

Thus,

* "score desc" requests to sort by score descendingly
* "score" would request to sort by score ascendingly
* "title" would request to sort by title ascendingly
* "title asc" would request to sorty by title ascendingly, too
* "title desc" would request to sort by title descendingly